### DISCUSSION TITLE
partial clones & sparse-checkout support

### DISCUSSION THREAD

**Discussion by erinmez** [community/community#5832] [Codespaces]
Hi there

First of all, Codespaces is such a wonderful product, thanks so much for bringing this to life  ❤️
 
We are introducing Bazel into our organisation and are consolidating into 1 monorepo which will be unwieldy by nature. Locally, partial clones & sparse-checkout (via scalar) works really well for us. Therefore, I would like to request support for partial clones & sparse-checkout in Codespaces too.

Our use case is, that when a reviewer on a PR opens up a Codespace, we would like to have just the relevant folders/files ready in the reviewer's Codespace. This is to speed up the physical git operations and more importantly to cut down on the time the subsequent steps (prepare build, IDE loading/indexing of files) are minimized to the absolute necessary.

A minimal solution could be that before Codespaces does git checkout, a step calls out to our build service (via curl) with some metadata such PR number, branch and requested checkout revision. From there we return a list of directories (e.g. foo/moduleA bar/moduleB). This result would then influence the sparse-checkout and thus limit the physical data load on the Codespace (pseudo sequence as follows).

```
git clone --filter=blob:none --no-checkout  .../foobar

cd foobar

git sparse-checkout init --cone

git checkout <branch where the PR was created on>

git sparse-checkout set foo/moduleA bar/moduleB

```

Best Tanju

**Comment by timbussmann** [comment]
Functionality like this would also be very interesting when having a samples/documentation repository that contains various samples in different folders. You want users to be able to checkout and run a specific sample (rather than all the samples) via a template.
